### PR TITLE
closed #340

### LINKED ISSUE #340: Add the initial destinations features
Create a destination by analyzing a Colab or Jupyter activity 
Create a destination by analyzing a path containing Colab and/or Jupyter activities. 
Allow users to save a destination by providing a title for it. The JSON should be saved to a readable location in the Firebase database called /destinations/$destinationKey -> {$ownerKey:true}

Destinations are a JSON object containing a collection of skills and a name. Destinations do not need to reference the path or activity that they were generated from. However, it should be fine to include some metadata that could optionally be saved. The originating path, activity, origination time, and originating users are all interesting properties that could be saved along with the destination. 
$destinationKey -> { skills: SKILLSOBJ, updatedOn: SERVERTIME,  metadata: {orginanator, sourcePathKey, sourdeActivityKey, etc.} }

Destinations will only be created by analyzing a single activity or single path of activities. 

Only path owners can create / update the destination for a path. However, we could choose to make these destination objects publicly readable for public paths. Then users could navigate to the last updated destination for the path.

### PR DESCRIPTION
**Pull request by walkwel**

### CODE REVIEW

**Comment by walkwel** [comment]
Routes for Destination
/destinations
/destinations/<destinationKey>
/my-destinations

**Comment by walkwel** [comment]
Routes for Destination
/destinations
/destinations/destinationKey
/my-destinations

File: database.rules.json [hunk 1]
```diff
@@ -135,6 +135,10 @@
         "
       }
     },
+    "destinations":{
```
Comment (FeynmanDNA) [inline database.rules.json]: @scboesch please review the database rules

Comment (Brainenjii) [reply database.rules.json]: These rules don't allow removing destinations if I get it right. Do we need to remove them? Also, is there a typo in `orginanator`  field?

Comment (walkwel) [reply database.rules.json]: @scboesch kindly suggest.

Comment (walkwel) [reply database.rules.json]: Here are the firebase rules for destinations
```
 "destinations":{
      ".read" : "auth != null",
      "$destinationKey" :{
        ".read" : "auth != null",
        ".write" : "
            auth != null &&  
            (
              ( 
                newData.exists() && 
                newData.child('originator').exists() &&
                newData.child('originator').val() == auth.uid && 
                ( 
                  !data.exists()  || 
                  ( data.child('originator').val() == auth.uid) 
                ) 
              ) || ( 
                data.exists() && data.child('originator').val() == auth.uid 
              )
            )"
        }
    },
```

**Pull request merged**